modules/indexer/code/indexer.go

```diff
@@ -63,33 +87,59 @@ func Init() {
 	waitChannel := make(chan time.Duration)
 	go func() {
 		start := time.Now()
-		log.Info("PID: %d Initializing Repository Indexer at: %s", os.Getpid(), setting.Indexer.RepoPath)
-		defer func() {
-			if err := recover(); err != nil {
-				log.Error("PANIC whilst initializing repository indexer: %v\nStacktrace: %s", err, log.Stack(2))
-				log.Error("The indexer files are likely corrupted and may need to be deleted")
-				log.Error("You can completely remove the %q directory to make Gitea recreate the indexes", setting.Indexer.RepoPath)
+		var populate bool
+		switch setting.Indexer.RepoType {
+		case "bleve":
+			log.Info("PID: %d Initializing Repository Indexer at: %s", os.Getpid(), setting.Indexer.RepoPath)
+			defer func() {
```

Comment:
Can we really not use something like the CreateQueue/NewQueue or other registered intialiser route?

Reply:
The changes about the queues will be another PR to not make this one too big since it has been stayed here too long.